tests/core/test_stuff.py

```diff
@@ -355,3 +355,36 @@ def test_categorize_step():
         steps2 = mass2.misc.unpickle_object(tmpfilename)
         assert len(steps2) == 1
         assert isinstance(steps2[0][0], mass2.core.cal_steps.CategorizeStep)
+
+
+def test_external_trigger_experiment_state():
+    off_paths = mass2.core.ljhutil.find_ljh_files(pulsedata.off["ebit_20240723_0000"], ext=".off")
+    assert len(off_paths) == 2
+
+    data = mass2.Channels.from_off_paths(off_paths, "ebit_20240723_0000").with_experiment_state_by_path()
+
+    # Check that the experiment states are in the order and number we expect
+    series = data.channels[3].df["state_label"]
+    labels = series.unique()
+    counts = series.unique_counts()
+    expect_labels = ("START", "IGNORE", "B", "C", "D", "E", "F", "G")
+    expect_counts = (54669, 22957, 192, 398, 3790, 3947, 2284, 3853)
+    for L, C, eL, eC in zip(labels, counts, expect_labels, expect_counts):
+        assert L == eL
+        assert C == eC
+
+    # Now load and check external trigger file
+    dir = os.path.dirname(off_paths[0])
+    trigfile = os.path.join(dir, "20240723_run0000_external_trigger.bin")
+    data = data.with_external_trigger_by_path(trigfile)
+    ch = data.channels[3]
+
+    sprev = ch.df["subframecount_prev_ext_trig"]
```

Comment:
How about "subframe_count_to_prev_ext_trig", I find the current name ambiguous.

Reply:
Ambiguous how? It's not a count _to_ the previous ET. It's the count _of_ the previous ET. This seems clearer to me, and differences are easily computed if that's what you want.